File: react/src/client/scripts/models/Recipe.js
/**
 * Represents cookbook recipe.
 * @constructor
 */
define(['lib/helpers'], function (helpers) {

    function Recipe() {
        this.id = helpers.uid();
        this.name = '';
        this.description = '';
        this.photoId = '';
        this.popularity = 0; // [1, 2, 3, ..., 10]
        this.complexity = 0; // [1, 2, 3, ..., 10]
        this.ingredients = []; // string[]
        this.steps = []; // string[]
    }
    
    Recipe.prototype.equals = function(other) {
      return JSON.stringify(this) === JSON.stringify(other);
    };
    
    Recipe.prototype.clone = function() {
      var recipe = new Recipe();
      var copy = JSON.parse(JSON.stringify(this));
      Object.getOwnPropertyNames(copy)
            .forEach(function(prop) { recipe[prop] = copy[prop] });
      return recipe;
    };
    
    return Recipe;
});

File: go+vue/src/server/model/Recipe.go
package model

import (
	"encoding/json"
)

// Recipe model represents cooking recipe
type Recipe struct {
	ID          int             `json:"id"`
	Name        string          `json:"name"`
	Description string          `json:"description"`
	Complexity  int             `json:"complexity"`
	Popularity  int             `json:"popularity"`
	Ingredients json.RawMessage `json:"ingredients"`
	Steps       json.RawMessage `json:"steps"`
}
